timbermill-local/src/main/java/com/datorama/oss/timbermill/ElasticsearchClient.java

```diff
@@ -671,62 +678,54 @@ long countByName(String name, String env) throws IOException {
         return countResponse.getCount();
     }
 
-	public void retryFailedRequestsFromMemory() {
-		if (!failedRequests.isEmpty()) {
-			LOG.info("------------------ Failed Requests From Memory Retry Start ------------------");
-			List<Pair<DbBulkRequest, Integer>> list = memoryFailedRequestsAsList();
-			for (Pair<DbBulkRequest, Integer> entry : list) {
-				DbBulkRequest dbBulkRequest = entry.getKey();
-				Integer retryNum = entry.getValue();
-				sendDbBulkRequest(dbBulkRequest, retryNum + 1);
+	boolean handleBulkRequestFailure(DbBulkRequest dbBulkRequest, int tryNum, BulkResponse responses, String failureMessage) {
+		boolean stopTrying = false;
+		LOG.warn("Try number {}/{} for requests of size {} has failed, failure message: {}.", tryNum, numOfElasticSearchActionsTries, dbBulkRequest.getRequest().estimatedSizeInBytes(),
+				failureMessage);
+
+		if (failureMessage != null && failureMessage.contains("type=null_pointer_exception")) {
+			// there is NPE, need to stop retries
+			DbBulkRequest failedRequest = extractFailedRequestsFromBulk(dbBulkRequest, responses);
+			LOG.error("Null Pointer Exception Error in script. Requests:");
+			failedRequest.getRequest().requests().forEach(r -> LOG.error(r.toString()));
+			stopTrying = true;
+		}
+
+		else if (tryNum == numOfElasticSearchActionsTries + 1) {
```

Comment:
why not "tryNum> numOfElasticSearchActionsTries" ?

Reply:
+1